static/app/components/nav/utils.tsx

```diff
@@ -64,61 +126,6 @@ export interface NavSubmenuItem extends NavItem {
 
 export type NavConfig = NavItemLayout<NavSidebarItem>;
 
-export type NavigationItemStatus = 'inactive' | 'active' | 'active-parent';
-
-/**
- * Determine if a given SidebarItem or SubmenuItem is active
- */
-export function isNavItemActive(
-  item: NavSidebarItem | NavSubmenuItem,
-  location: ReturnType<typeof useLocation>
-): boolean {
-  const to = resolveNavItemTo(item);
-  if (!to) {
-    return false;
-  }
-
-  /**
-   * Issue submenu is special cased because it is matched based on query params
-   * rather than the pathname.
-   */
-  if (location.pathname.includes('/issues/') && to.includes('/issues/')) {
-    const {label} = item;
-    const matches = hasMatchingQueryParam({to, label}, location);
-    const isDefault = label === 'All';
-    if (location.search) {
-      return matches || isDefault;
-    }
-    return isDefault;
-  }
-
-  const normalizedTo = normalizeUrl(to);
-  const normalizedCurrent = normalizeUrl(location.pathname);
-  // Shortcut for exact matches
-  if (normalizedTo === normalizedCurrent) {
-    return true;
-  }
-  // Fallback to legacy nav logic
-  return isItemActive({to, label: item.label});
-}
-
-export function isSubmenuItemActive(
-  item: NavSidebarItem,
-  location: ReturnType<typeof useLocation>
-): boolean {
-  if (!item.submenu) {
-    return false;
-  }
-  if (isNonEmptyArray(item.submenu)) {
-    return item.submenu.some(subitem => isNavItemActive(subitem, location));
-  }
-  return (
-    item.submenu.main.some(subitem => isNavItemActive(subitem, location)) ||
-    item.submenu.footer?.some(subitem => isNavItemActive(subitem, location)) ||
-    false
-  );
-}
-
```

Comment:
Rather than introducing complex logic to derive the state of primary and secondary nav items from the config, this PR now tracks state and exposes it via context.